src/uwtools/file.py

```diff
@@ -54,6 +54,19 @@ def __init__(
         )
         self._validate()
 
+    def _check_dst_paths(self, cfg: dict[str, str]) -> None:
+        """
+        Check that all destination paths are absolute if no target directory is specified.
+
+        :parm cfg: The dst/linkname -> src/target map.
+        :raises: UWConfigError if no target directory is specified and a relative path is.
+        """
+        if not self._target_dir:
+            errmsg = "Relative path '%s' requires the target directory to be specified"
+            for dst in cfg.keys():
+                if not Path(dst).is_absolute():
+                    raise UWConfigError(errmsg % dst)
```

Comment:
This method does the checking for compatibility between the presence/absence of `target_dir` and the absolute/relative status of destination/link paths.